test/fluture.test.js

```diff
@@ -459,6 +460,140 @@ describe('Constructors', () => {
 
   });
 
+  describe('.do()', () => {
+
+    it('throws TypeError when not given a function', () => {
+      const xs = [NaN, {}, [], 1, 'a', new Date, undefined, null];
+      const fs = xs.map(x => () => Future.do(x));
+      fs.forEach(f => expect(f).to.throw(TypeError, /Future/));
+    });
+
+    it('throws TypeError when the given function does not return an interator', () => {
+      const xs = [NaN, {}, [], 1, 'a', new Date, undefined, null, () => {}, {next: 'hello'}];
+      const fs = xs.map(x => () => Future.do(() => x).fork(noop, noop));
+      fs.forEach(f => expect(f).to.throw(TypeError, /Future/));
+    });
+
+    it('throws TypeError when the returned iterator does not return a valid iteration', () => {
+      const xs = [null, '', {}, {done: true}, {value: 1, done: 1}];
+      const fs = xs.map(x => () => Future.do(() => ({next: () => x})).fork(noop, noop));
+      fs.forEach(f => expect(f).to.throw(TypeError, /Future/));
+    });
+
+    it('throws TypeError when the returned iterator produces something other than a Future', () => {
+      const xs = [NaN, {}, [], 1, 'a', new Date, undefined, null];
+      const fs = xs.map(x => () =>
+        Future.do(() => ({next: () => ({done: false, value: x})})).fork(noop, noop)
+      );
+      fs.forEach(f => expect(f).to.throw(TypeError, /Future/));
+    });
+
+    it('can be used to chain Futures in do-notation', () => {
+      const actual = Future.do(function*(){
+        const a = yield Future.of(1);
+        const b = yield Future.of(2);
+        return a + b;
+      });
+      return Promise.all([
+        assertResolved(actual, 3),
+        assertResolved(actual, 3)
+      ]);
+    });
+
+    it('is stack safe', () => {
+      const gen = function*(){
+        let i = 0;
+        while(i < STACKSIZE + 1) yield Future.of(i++);
+        return i;
+      };
+      const m = Future.do(gen);
+      return assertResolved(m, STACKSIZE + 1);
+    });
+
+  });
+
+  describe('.chainRec()', () => {
```

Comment:
👍 looks like @rpominov has good tests [here](https://github.com/rpominov/fun-task/blob/master/test/chainRec.js). Also [this](https://github.com/ramda/ramda-fantasy/blob/master/test/future.test.js#L106-L148) is what I have added to ramda-fantasy. You can adopt some of those.